nucleus/cluster/cli/src/main/java/com/sun/enterprise/admin/cli/cluster/ChangeNodeMasterPasswordCommand.java

```diff
@@ -188,6 +189,22 @@ protected void createMasterPasswordFile() throws CommandException {
     }
 
 
+    protected void chmod(String args, File file) throws IOException {
+        if (OS.isUNIX()) {
+            if (!file.exists()) throw new IOException(Strings.get("fileNotFound", file.getAbsolutePath()));
+
+            // " +" regular expression for 1 or more spaces
+            final String[] argsString = args.split(" +");
+            List<String> cmdList = new ArrayList<String>();
+            cmdList.add("/bin/chmod");
+            for (String arg : argsString)
+                cmdList.add(arg);
+            cmdList.add(file.getAbsolutePath());
+            new ProcessBuilder(cmdList).start();
```

Comment:
This chmod implementation is problematic for multiple reasons :disappointed: 
This code just makes and relies upon too many assumptions.

If it's only used to set mode 0600 on files... the same thing can be achieved with Java7 NIO2 or maybe even Java6 setReadable/setWritable method 2-parameter variants which specify to only make it readable and writable for the owner, but neither by non-owners.